### ISSUE TITLE
import_yaml/pillar.get override with empty defaults.yml fails on 2016.11.2

### ISSUE THREAD

**Issue by lomeroe** [saltstack/salt#39062]
### Description of Issue/Question
Attempting to import_yml an empty defaults.yml and then use as a pillar.get override throws an error in 2016.11.2.

For example, a state created with something like the following (our state/formula cookie-cutter does this for every state we create):

```
{% import_yaml 'path/to/formula/defaults.yml' as formula_defaults -%}
{% from 'path/to/formula/map.jinja' import formula with context %}

{% set formula_settings = salt['pillar.get']('formula:pillar', default=formula_defaults, merge=True) -%}
```

Starting with 2016.11.2, when using the above setup, if the defaults.yml is empty, the following error is now thrown:
```
Problem running salt function in Jinja template: default must be a dictionary when merge=True
```
Oddly (at least to me), state.show_sls does not throw an error.

Perhaps this is to be expected (which is fine), but had functioned previously.  The workaround is to put something in the defaults.yml:

```
empty_defaults: True
```

### Setup
Create a state using the defaults/pillar override block as described above, leave defaults.yml empty.

### Steps to Reproduce Issue
attempt to apply state created as described

### Versions Report
Issue seen on RHEL6/7 and CentOS 7

########## RHEL6
Salt Version:
           Salt: 2016.11.2

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 1.4.1
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.8.1
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: 0.20.2
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.6.6 (r266:84292, Aug  9 2016, 06:11:56)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.5.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5

#######CentOS 7
Salt Version:
           Salt: 2016.11.2

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.5 (default, Nov  6 2016, 00:28:07)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.3.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: centos 7.3.1611 Core
        machine: x86_64
        release: 3.10.0-327.10.1.el7.x86_64
         system: Linux
        version: CentOS Linux 7.3.1611 Core

**Comment by gtmanfred** [comment]
I have confirmed this change in behavior.

Thanks for reporting,
Daniel

**Comment by gtmanfred** [comment]
@lomeroe i believe that #39116 should fix this problem for you?

Thanks,
Daniel

**Comment by terminalmage** [comment]
Fixed in https://github.com/saltstack/salt/issues/39062, with bonus unit test included.

**Comment by pirxthepilot** [comment]
I'd like to report that this behavior is also present in 2016.3.5, CentOS 6 and 7.  We did not get this error on the previously installed version which was 2016.3.2.

**Comment by rallytime** [comment]
@pirxthepilot The fix was made against the 2016.3 branch because @terminalmage is on the ball. :D It will be merged forward to the 2016.11 branch as well. Thanks for the update!

**Comment by pirxthepilot** [comment]
@rallytime Yay! Thanks so much!
